dbus/dbus.c

```diff
@@ -43,6 +43,9 @@ bool init_dbus(struct mako_state *state) {
 }
 
 void finish_dbus(struct mako_state *state) {
+	while ((sd_bus_process(state->bus, NULL)) > 0) {
```

Comment:
According to the [docs](https://www.freedesktop.org/software/systemd/man/sd_bus_default.html), it's a good idea to call `sd_bus_flush` followed by `sd_bus_close` before dropping the ref (the former will flush outgoing messages, the latter drop incoming messages).

Reply:
Found a better way, I think: https://github.com/systemd/systemd/pull/485

Annoyingly undocumented though.